src/utils/tree/getNodeTags/getNodeTags.ts

```diff
@@ -0,0 +1,30 @@
+import type { TreeNode } from "@/features/Treege/type/TreeNode";
+
+/**
+ * Get node names from tree
+ * @param tree
+ */
+const getNodeTags = (tree: TreeNode | null): string[] => {
+  if (!tree) {
+    return [];
+  }
+
+  const nodeTags: Set<string> = new Set();
+
+  const extractNodeNames = (node: TreeNode) => {
+    if (node?.attributes?.tag) {
+      nodeTags.add(node?.attributes?.tag);
```

Comment:
Ici tu peux supprimer l'optional chaining et faire directement :

```suggestion
      nodeTags.add(node.attributes.tag);
```
car tu test déjà si il y un tag `if (node?.attributes?.tag) {`